browser/brave_stats_tab_helper.cc

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_stats_tab_helper.h"
+
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/browser/brave_stats_updater.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace brave {
+
+BraveStatsTabHelper::BraveStatsTabHelper(content::WebContents *web_contents)
+    : content::WebContentsObserver(web_contents) {}
+
+BraveStatsTabHelper::~BraveStatsTabHelper() = default;
+
+void BraveStatsTabHelper::DidStartNavigation(
+        content::NavigationHandle* handle) {
+  if (!handle || !handle->IsInMainFrame() || handle->IsDownload())
+    return;
+
+  auto transition = handle->GetPageTransition();
+
+  switch (ui::PageTransitionStripQualifier(transition)) {
+  case ui::PAGE_TRANSITION_TYPED:
```

Comment:
you're missing a header for the transition types (and also the corresponding dep for `//ui/base`